components/admission-webhook/Dockerfile

```diff
@@ -10,12 +10,8 @@ COPY . .
 ENV GO111MODULE=on
 
 # Build
-RUN if [ "$(uname -m)" = "aarch64" ]; then \
-        CGO_ENABLED=0 GOOS=linux GOARCH=arm64 go build -o webhook -a . ; \
-    else \
-        CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -o webhook -a . ; \
-    fi
-
+RUN CGO_ENABLED=0 GOOS=linux go build -o webhook -a . ;
```

Comment:
I understand that by omitting the GOARCH env var we let golang decide on the architecture based on the node environment (machine). 

I'm just trying to wrap my mind around the next step. Is it going to be the process described in https://docs.docker.com/build/building/multi-platform/ to build images that have manifests for different platforms?

Reply:
Yes, it's good to let GO determine the arch, so we don't have to maintain an arch list explicitly here or wrap around boiler-plate code with arch-specific if/else statements.

Instead, we now shift the control which arch is actually build to the build system. If doing nothing special, the arch of the build machine is simply used (and as Kubeflow is currently build on amd64, we are backwards-compatible to the current behavior).

In further PRs, we will additionally modify docker-based builds using buildx, where you can, for instance, do something like this:
`docker buildx build --platform linux/amd64,linux/ppc64le ... `

Here, docker will automatically run actually 2 builds: one for amd64 and one for ppc64le. When coming to above GO-code, GO will acknowledge the external platform configuration and build it correctly. In case no native hardware is available for the given platform, docker will emulate the architecture using QEMU - so you can also build for different archs on amd64.

The final outcome is a single multi-arch image with support of all archs listed in Docker's platform statement.

Reply:
@lehrig this was a *very* thorough explanation of the suggested approach! I'd also suggest to cross post it in the umbrella issue, so that it's readily available for anyone wondering how to e2e approach is going to be.

Also LGTM